firedrake/assemble.py

```diff
@@ -296,9 +301,9 @@ def get_map(x, bcs=None, decoration=None):
                 # In the case of extruded meshes with horizontal facet integrals, two
                 # parallel loops will (potentially) get created and called based on the
                 # domain id: interior horizontal, bottom or top.
-                index, itspace = m.exterior_facets.measure_set(integral_type, subdomain_id)
-                decoration = index
-                kwargs["iterate"] = index
+                decoration = {"exterior_facet_top": op2.ON_TOP,
+                              "exterior_facet_bottom": op2.ON_BOTTOM}[integral_type]
+                kwargs["iterate"] = decoration
```

Comment:
I've moved the "decoration" bit out of the measure_set method to here (which is the only place it's used).